### ISSUE TITLE
nn.LayerNorm after nn.MultiheadAttention

### ISSUE THREAD

**Issue by wonss737** [facebookresearch/detr#355]
The output shape of nn.MultiheadAttention is (L, N, E) (where L is the target sequence length, N is the batch size, E is the embedding dimension.) and this tensor enter into nn.LayerNorm.

But, the input shape of nn.LayerNorm should be (N, *), so I think the output of nn.MultiheadAttention should be permuted before nn.LayerNorm. Is it right?

**Comment by alcinos** [comment]
In the encoder (https://pytorch.org/docs/1.6.0/_modules/torch/nn/modules/transformer.html#TransformerEncoderLayer), the LayerNorm are created as ```LayerNorm(d_model)```, which, according to the LN doc (https://pytorch.org/docs/1.6.0/generated/torch.nn.LayerNorm.html?highlight=layernorm#torch.nn.LayerNorm) means that only the last dimension will be normalized, and it doesn't matter in what order the other dimensions are.

I believe I have answered your question and as such I'm closing this. Feel free to reach out if you have further concerns.
